app/models/user.rb

```diff
@@ -122,7 +135,66 @@ def assign_barcode
     return self.barcode
   end
 
-  
+
+  def check_barcode_uniqueness_with_sierra(barcode_to_check)
+    # Ask the platform microservice api to ask Sierra if there is already a user
+    # with the passed-in barcode.
+    # Most users will have an equivalent user record in Sierra, if set up correctly.
+    # New users and users whose records have been purged from Sierra might not.
+    # Return "true" if a user is found, false otherwise.  Default to "false".
+    # Throw an exception if called with malformed data.
+
+    if barcode_to_check.blank?
+      return false
+    end
+
+    @barcode_found_in_sierra = false
+
+    response = HTTParty.get(
+      ENV['PATRON_MICROSERVICE_URL_V01'] + "?barcode=#{barcode_to_check}",
+      headers:
+        { 'Authorization' => "Bearer #{Oauth.get_oauth_token}",
+          'Content-Type' => 'application/json' },
+      timeout: 10
+    )
+    case response.code
+    when 200
+      @barcode_found_in_sierra = true
+      LogWrapper.log('DEBUG', {
+          'method' => "#{model_name}.check_barcode_uniqueness_with_sierra",
+          'message' => "patron service found user(#{barcode_to_check}) in Sierra",
+          'status' => response.code
+        })
+    when 404
+      LogWrapper.log('DEBUG', {
+          'method' => "#{model_name}.check_barcode_uniqueness_with_sierra",
+          'message' => "patron service did not find user(#{barcode_to_check}) in Sierra",
+          'status' => response.code
+        })
+    when 409
+      # Duplicate patrons found for query.  This is a data cleanliness/sync problem
+      # but for the purpose of this method, we just care that the barcode is taken.
+      @barcode_found_in_sierra = true
+      LogWrapper.log('INFO', {
+          'method' => "#{model_name}.check_barcode_uniqueness_with_sierra",
+          'message' => "patron service found multiple user(#{barcode_to_check}) records in Sierra",
+          'status' => response.code
+        })
+    else
+      # Includes response of 500.  Be liberal and assume the barcode is free.
+      LogWrapper.log('ERROR', {
+          'method' => "#{model_name}.check_barcode_uniqueness_with_sierra",
+          'message' => "patron service threw error while looking for user(#{barcode_to_check}) in Sierra",
+          'status' => response.code,
+          'responseData' => response.body
+        })
+      raise Exceptions::InvalidResponse, "Invalid status code of: #{response.code}"
+    end
+
+    return @barcode_found_in_sierra
```

Comment:
Better to return status code also. Like this [@barcode_found_in_sierra, response.code] .Because status code 200 and 409 in both cases return's  this value is true(@barcode_found_in_sierra = true). Caller method do not know about the status.  If status code is 200 means different functionality and status code 409 means different functionality.

Reply:
I see what you're saying, but for now, I'd like to keep the business logic encapsulated in this method.  Let's see if we'll need to do more sophisticated response processing outside of this method.